Makefile

```diff
@@ -119,27 +120,27 @@ generate: generate-code kubefedctl
 
 push: container
 
-	if [ -z "$(TRAVIS_PULL_REQUEST)" ]; \
+	if [[ -z "$(TRAVIS_PULL_REQUEST)" ]]; \
 	then \
 		$(DOCKER) push $(REGISTRY)/$(TARGET):$(GIT_VERSION); \
-	elif [ "$(TRAVIS_PULL_REQUEST)" = "false" ]; \
+	elif [[ "$(TRAVIS_PULL_REQUEST)" == "false" && "$(TRAVIS_SECURE_ENV_VARS)" == "true" ]]; \
 	then \
 		$(DOCKER) login -u "$(QUAY_USERNAME)" -p "$(QUAY_PASSWORD)" quay.io; \
-		if [ "$(TRAVIS_BRANCH)" = "master" ]; \
+		if [[ "$(TRAVIS_BRANCH)" == "master" ]]; \
 		then \
 			$(DOCKER) tag $(REGISTRY)/$(TARGET):$(GIT_VERSION) $(REGISTRY)/$(TARGET):canary; \
 			$(DOCKER) push $(REGISTRY)/$(TARGET):canary; \
+		fi; \
+		\
+		if git describe --tags --exact-match >/dev/null 2>&1; \
+		then \
+			$(DOCKER) tag $(REGISTRY)/$(TARGET):$(GIT_VERSION) $(REGISTRY)/$(TARGET):$(GIT_TAG); \
```

Comment:
(suggestion) Not related to this pr, but can we use $(IMAGE) which is defined at the top of Makefile instead of $(REGISTRY)/$(TARGET)

Reply:
:+1: I've updated.